File: solutions_1-10/Euler_7.py
# What is the 10001st prime number?

import math
import timeit


start = timeit.default_timer()

def is_prime(n):
    if n == 2:
        return True
    if n == 1 or n % 2 == 0:
        return False
    for i in xrange(3, int(math.sqrt(n) + 1), 2):
        if n % i == 0:
            return False
    return True

def prime_gen():
    number_prime = 0
    n = 0
    while number_prime < 10001:
        if is_prime(n):
            number_prime += 1
            yield n
        n += 1

def euler_7():
    prime = prime_gen()
    while True:
        try:
            answer = prime.next()
        except StopIteration:
            return answer

print "Answer: %s" % euler_7()
stop = timeit.default_timer()
print "Time: %f" % (stop - start)    

File: Java/java_solutions_1-10/Euler_7.java
/*
/   Project Euler problem #7 10001 prime
/
/   What is the 10001 prime number?
/
*/

import java.util.stream.IntStream;
import java.lang.Math;


public class Euler_7
{
    public static boolean isPrime(int n)
    {
        if (n < 2) return false;
        else if (n == 2) return true;
        int limit = (int) Math.sqrt(n);
        return IntStream.range(3, limit + 1)
                        .allMatch(x -> n % x != 0);
    }

    public static void main(String[] args)
    {
        long start = System.nanoTime();
        int i = 3;
        for (int primeCount=1; primeCount < 10001; i+=2)
            if (isPrime(i)) primeCount++;

        System.out.println("Answer: " + (i - 2));
        long stop = System.nanoTime();
        System.out.printf("Time: %.4f\n", ((float) stop - start) / 1_000_000_000);
    }
}
